tracer/src/Datadog.Trace/Iast/SensitiveData/JsonTokenizer.cs

```diff
@@ -0,0 +1,85 @@
+// <copyright file="JsonTokenizer.cs" company="Datadog">
+// Unless explicitly stated otherwise all files in this repository are licensed under the Apache 2 License.
+// This product includes software developed at Datadog (https://www.datadoghq.com/). Copyright 2017 Datadog, Inc.
+// </copyright>
+
+using System.Collections.Generic;
+using System.IO;
+using System.Text.RegularExpressions;
+using Datadog.Trace.Configuration;
+using Datadog.Trace.Vendors.Newtonsoft.Json;
+
+#nullable enable
+
+namespace Datadog.Trace.Iast.SensitiveData;
+
+internal class JsonTokenizer : ITokenizer
+{
+    private static readonly Regex SourceValueRegex = new(@"(?i)bearer\s+[a-z0-9\._\-]+|token:[a-z0-9]{13}|gh[opsu]_[0-9a-zA-Z]{36}|ey[I-L][\w=-]+\.ey[I-L][\w=-]+(\.[\w.+\/=-]+)?|[\-]{5}BEGIN[a-z\s]+PRIVATE\sKEY[\-]{5}[^\-]+[\-]{5}END[a-z\s]+PRIVATE\sKEY|ssh-rsa\s*[a-z0-9\/\.+]{100,}", RegexOptions.Compiled);
+
+    public List<Range> GetTokens(string value, IntegrationId? integrationId = null)
+    {
+        var redactedRanges = new List<Range>();
+
+        // Remove new lines from the value by space
+        // to get the correct position of the token with the .LinePosition property
+        value = value.Replace("\n", " ");
+
+        using var sr = new StringReader(value);
+        using var reader = new JsonTextReader(sr);
+
+        while (reader.Read())
+        {
+            switch (reader.TokenType)
+            {
+                case JsonToken.String:
+                case JsonToken.Integer:
+                case JsonToken.Float:
+                case JsonToken.Boolean:
```

Comment:
Are we always redacting all the values? Even integers, float or bools?

Reply:
That was a question asked in the RFC: https://docs.google.com/document/d/1DYqoY5HMwKOQzzJhRXJCKO5p8pW9heKueUzFnoL8iRk/edit?pli=1#heading=h.lq3n19z70wbp

The term 'literals' is used to refer to all typeof values. According to the comments, it seems that the Agent is also redacting all literals.